auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java

```diff
@@ -64,7 +74,17 @@
   private Map<String, List<String>> lastMetadata;
 
   public GoogleAuthLibraryCallCredentials(Credentials creds) {
-    this.creds = checkNotNull(creds, "creds");
+    checkNotNull(creds, "creds");
+
+    // Avoid instanceof since that requires the class to exist. Only after we know there's an
+    // instance of the class is it reasonably safe to assume the class exists.
+    for (Class<?> clazz = creds.getClass(); clazz != null; clazz = clazz.getSuperclass()) {
```

Comment:
Yes, although then that means the class has to be available initially. Right now it is loading as late as possible on purpose. I guess it'd be better to be limited and expand the capabilities when necessary.